.travis.yml

```diff
@@ -1,22 +1,77 @@
 language: bash
-sudo: required
-script: bash shellcheck && sudo bash buildall
-dist: xenial
+dist: focal
+virt: vm
+group: edge
+os: linux
 services:
   - docker
-before_install:
-  - docker version
-  # Fix for Ubuntu Xenial apt-daily.service triggering
-  # https://unix.stackexchange.com/questions/315502/how-to-disable-apt-daily-service-on-ubuntu-cloud-vm-image
-  - |
-    while sudo fuser /var/{lib/{dpkg,apt/lists},cache/apt/archives}/lock >/dev/null 2>&1; do
-      sleep 1
-    done
-  - sudo apt-get -qq update
-  - sudo apt-get install -y debian-archive-keyring debootstrap shellcheck
-deploy:
-  provider: script
-  script: bash pushall
-  skip_cleanup: true
-  on:
-    branch: master
+
+.build_job: &build_job
+  stage: build
+  before_install:
+    - docker version
+    # Fix for Ubuntu Xenial apt-daily.service triggering
+    # https://unix.stackexchange.com/questions/315502/how-to-disable-apt-daily-service-on-ubuntu-cloud-vm-image
+    - |
+      while sudo fuser /var/{lib/{dpkg,apt/lists},cache/apt/archives}/lock >/dev/null 2>&1; do
+        sleep 1
+      done
+    - sudo rm /usr/local/bin/jq
+  install:
+    - sudo make .installed-requirements
+  script: sudo bash buildone $DIST $PLATFORM
+  after_success:
+    - 'if [[ "$TRAVIS_BRANCH" == "master" && "$DIST" == "buster" ]] ; then sudo docker tag "bitnami/minideb:$DIST-$PLATFORM" "$BASENAME:latest-$PLATFORM" ; fi'
+    - 'if [[ "$TRAVIS_BRANCH" == "master" ]] ; then sudo bash pushone $DIST $PLATFORM ; fi'
+    - 'if [[ "$TRAVIS_BRANCH" == "master" && "$DIST" == "buster" ]] ; then sudo bash pushone latest $PLATFORM ; fi'
+
+jobs:
+  include:
+    - stage: shellcheck
+      install:
+        - sudo apt-get -qq update
+        - sudo apt-get install -y shellcheck
+      script: bash shellcheck
+    - <<: *build_job
+      arch: amd64
+      env:
+        - DIST=jessie PLATFORM=amd64
+    - <<: *build_job
+      arch: amd64
+      env:
+        - DIST=stretch PLATFORM=amd64
+    - <<: *build_job
+      arch: amd64
+      env:
+        - DIST=buster PLATFORM=amd64
+    - <<: *build_job
+      arch: arm64-graviton2
+      env:
+        - DIST=stretch PLATFORM=arm64
```

Comment:
Didn't notice that build was running on `.org`. Fixed

Reply:
Anyway on `.org` now there's a warning stating that will be shut down in a few weeks. If the build will be migrated to `.com` maybe we should revert to `graviton2` which should have better performance

Reply:
There could be a problem with `arm64` architecture on `travis-ci.org`: it could run only in _unpriviledged_ LXD container.
To correctly run `debootstrap` a full vm is needed, which is only available in `arm64-graviton2`.  
I also tried to execute a QEMU build into a full amd64 vm, but it is extremely slow.

Stating that `.org` is about to shut down, maybe a migration to `.com` could be planned?

Reply:
I wonder if `fakechroot fakeroot` would be enough to get this working.

Also there is https://github.com/josch/polystrap which might be useful only run on one arch.

Reply:
I tried with `fakechroot fakeroot` but it always fails.

The question is: is it worth it?

Travis announced the shut down of `.org` on 31st December 2020. Obviously it was not shut down, but can be off at any time now.

Additionally, travis com is not really free for OSS with the new business plans, and in forums there's plenty of users which used all the free credits and with no way to replenish them for free.

Travis builds could be a problem now.

Reply:
What is a bit strange is `arm64-graviton2` is working in `.org`, right?. It seems like they internally changed something and all the projects are behind those https://travis-ci.com/plans but using different URLs.

There are two options then:
- use `arm64-graviton2` and move this project to any of these plans
- use `qemu` to build these images even it takes so much time

Reply:
The third option is to try to build the images on GitHub Actions.
It shouldn't be too difficult to make a test workflow. The question is: will the execution be too slow?

Reply:
docker provides a qemu setup action for [multiarch GH workflows] (https://blog.oddbit.com/post/2020-09-25-building-multi-architecture-im/).

Im I'm not mistaken, GH by default runs up to eight jobs on eight runners concurrently (might be four not sure anymore) which [can be decreased](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymax-parallel) as well. all you'd need to do is define a job with a step that runs the qemu_build script with args `${{matrix.distro}} ${{matrix.other_arg}}`, then define your matrix on the top of the job and you should be all set for parallel builds :)

Reply:
For the records: I just tried to build the images on GH Actions on my fork (`master` branch to test the push job), without using `qemu_build` as performance was too bad.
[This is the last working run](https://github.com/alekitto/minideb/actions/runs/515972341).

Reply:
14 mins for buster arm64..

+1 for the effort though! plz dont delete the branch :)

Reply:
fyi I bumped the `build` job to `ubuntu-20.04` in the hope that this would run on newer github runners than the `18.04` job.

the [result](https://github.com/ddelange/minideb/actions/runs/516306433) is about twice as fast:
![image](https://user-images.githubusercontent.com/14880945/106107879-c4f69a00-6147-11eb-9cf9-36def81c9e7e.png)